src/main/java/org/cbio/gdcpipeline/tasklet/ProcessManifestFileTasklet.java

```diff
@@ -101,86 +101,31 @@ public RepeatStatus execute(StepContribution stepContribution, ChunkContext chun
         };
     }
 
-    /*  Payload format
-     * {
-     * "filters":{
-     *          "op":"in",
-     *          "content":{
-     *                  "field":"cases.case_id",
-     *                  "value":["ABC123"]
-     *                  }
-     *          },
-     * "format":"JSON",
-     * "fields":"file_name,cases.case_id,data_format",
-     *
-     * }
-     */
-
-    protected String buildJsonRequest() {
-        JsonObject node = new JsonObject();
-        JsonObject filters = new JsonObject();
-        JsonObject content = new JsonObject();
-        JsonArray values = new JsonArray();
-        for (ManifestFileData record : manifestFileList) {
-            values.add(record.getId());
-        }
-        content.addProperty("field", "file_id");
-        content.add("value", values);
-        filters.addProperty("op", "in");
-        filters.add("content", content);
-
-        node.add("filters", filters);
-        node.addProperty("format", "JSON");
-        node.addProperty("fields", "file_id,file_name,cases.case_id,type,data_format");
-        return node.toString();
-    }
-
-    protected void gdcApiRequest(String payload) throws Exception {
-        int callCount = 0;
-        int totalCallCount = 1;
-        int from = 0;
-
-        while (callCount < totalCallCount) {
-            callCount += 1;
-            String url = GDC_API_FILES_ENDPOINT + "?from=" + from + "&size=" + MAX_RESPONSE_SIZE;
-            if (LOG.isInfoEnabled()) {
-                LOG.info(" Calling GDC API : " + url);
-                LOG.info(" Payload : " + payload);
-            }
-            GdcApiResponse res = callGdcApi(url, payload);
-            totalCallCount = res.getData().getPagination().getPages();
-            if (totalCallCount == 0) {
-                if (LOG.isErrorEnabled()) {
-                    LOG.error("Last API Request returned 0 results");
+    private void processFilesResponse() {
+        JSONArray files = GraphQLQueryUtil.getQueryList(gdcResponse, "files");
+
+        for (Object edgeObject : files) {
+            JSONObject edge = (JSONObject) edgeObject;
+            // Get a list of all case ids from the manifest files
+            JSONObject node = (JSONObject) edge.get("node");
+            for (ManifestFileData manifestFileData : manifestFileList) {
```

Comment:
similar comment here as somewhere above - moving this json parsing/extraction somewhere else for clarity. Maybe a util class or something like that

Reply:
(can be a to-do for future refactoring)